proxy/session.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+type session struct {
+	IsActive    bool
+	MediatorURL string `json:"mediatorUrl"`
+}
+
+func sessionHandler(currentSession *session) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/session" {
+			return
+		}
+		if req.Method == "POST" {
+			startSession(currentSession, req)
```

Comment:
Can we have a race condition here?

Reply:
If someone tries to start an already starded session, we might have. I need to check `if req.Method == "POST" && currentSession.IsActive == false`.
Is this what you were thinking?